python/python/lancedb/pydantic.py

```diff
@@ -262,7 +268,7 @@ def _py_type_to_arrow_type(py_type: Type[Any], field: FieldInfo) -> pa.DataType:
         return pa.int64()
     elif py_type is float:
         return pa.float64()
-    elif py_type is str:
+    elif py_type is str or (StrEnum is not None and issubclass(py_type, StrEnum)):
         return pa.utf8()
```

Comment:
String could work fine here. Though I would think it would be preferable to use dictionary:

```suggestion
    elif py_type is str:
        return pa.utf8()
    elif StrEnum is not None and issubclass(py_type, StrEnum):
        return pa.dictionary(pa.int16(), pa.utf8())
```